servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java

```diff
@@ -50,72 +58,235 @@
 
-    @Override
+    /**
+     * Append the filter to the chain of filters used to decorate the {@link ConnectionFactory} used by this
+     * builder.
+     * <p>
+     * Filtering allows you to wrap a {@link ConnectionFactory} and modify behavior of
+     * {@link ConnectionFactory#newConnection(Object, TransportObserver)}.
+     * Some potential candidates for filtering include logging and metrics.
+     * <p>
+     * The order of execution of these filters are in order of append. If 3 filters are added as follows:
+     * <pre>
+     *     builder.append(filter1).append(filter2).append(filter3)
+     * </pre>
+     * Calling {@link ConnectionFactory} wrapped by this filter chain, the order of invocation of these filters will be:
+     * <pre>
+     *     filter1 ⇒ filter2 ⇒ filter3 ⇒ original connection factory
+     * </pre>
+     * @param factory {@link ConnectionFactoryFilter} to use.
+     * @return {@code this}.
+     */
     public abstract GrpcClientBuilder<U, R> appendConnectionFactoryFilter(
             ConnectionFactoryFilter<R, FilterableStreamingHttpConnection> factory);
 
-    @Override
+    /**
+     * Append the filter to the chain of filters used to decorate the {@link StreamingHttpConnection} created by this
+     * builder.
+     * <p>
+     * Filtering allows you to wrap a {@link StreamingHttpConnection} and modify behavior during request/response
+     * processing
+     * Some potential candidates for filtering include logging, metrics, and decorating responses.
+     * <p>
+     * The order of execution of these filters are in order of append. If 3 filters are added as follows:
+     * <pre>
+     *     builder.append(filter1).append(filter2).append(filter3)
+     * </pre>
+     * making a request to a connection wrapped by this filter chain the order of invocation of these filters will be:
+     * <pre>
+     *     filter1 ⇒ filter2 ⇒ filter3 ⇒ connection
+     * </pre>
+     *
+     * @param factory {@link StreamingHttpConnectionFilterFactory} to decorate a {@link StreamingHttpConnection} for the
+     * purpose of filtering.
+     * @return {@code this}.
+     */
     public abstract GrpcClientBuilder<U, R> appendConnectionFilter(StreamingHttpConnectionFilterFactory factory);
 
-    @Override
+    /**
+     * Append the filter to the chain of filters used to decorate the {@link StreamingHttpConnection} created by this
+     * builder, for every request that passes the provided {@link Predicate}.
+     * <p>
+     * Filtering allows you to wrap a {@link StreamingHttpConnection} and modify behavior during request/response
+     * processing
+     * Some potential candidates for filtering include logging, metrics, and decorating responses.
+     * <p>
+     * The order of execution of these filters are in order of append. If 3 filters are added as follows:
+     * <pre>
+     *     builder.append(filter1).append(filter2).append(filter3)
```

Comment:
To be consistent with HTTP, please use the real method name here: `appendConnectionFilter`.

Same for other `append*` methods we have below (pre-existing, I can not comment there).